topic-operator/src/main/java/io/strimzi/operator/topic/TopicSerialization.java

```diff
@@ -127,7 +129,28 @@ private static int getPartitions(KafkaTopic kafkaTopic) {
      */
     public static KafkaTopic toTopicResource(Topic topic, LabelPredicate resourcePredicate) {
         ResourceName resourceName = topic.getOrAsMapName();
-        return new KafkaTopicBuilder().withApiVersion("v1")
+        ObjectMeta om = topic.getMetadata();
+        if (om != null) {
```

Comment:
Just deal with the metadata in this if/else and put the rest of the code with the `KafkaTopicBuilder` outside the block. That will avoid duplicating the `KafkaTopicBuilder` code in both branches.